src/Native/Unix/CMakeLists.txt

```diff
@@ -11,6 +11,7 @@ set(VERSION_FILE_PATH "${CMAKE_BINARY_DIR}/../../version.c")
 
 add_compile_options(-Wno-format-nonliteral)
 add_compile_options(-Wno-disabled-macro-expansion)
+add_compile_options(-Wno-deprecated-declarations)
```

Comment:
Are there multiple projects that need this?  If it's just the crypto shim then I'd expect this to be in that specific file.

Reply:
Currently there is only one place that are defining these; in this file. To be consistent with the current norm, should it be kept here?
When the second project will require it, nobody will remember that it is also in a project-specific file to move at global location.

--- 

The second one-off approach is in the .c file, such as:

https://github.com/dotnet/corefx/blob/44e5bdafb8d989a220c9cf1b94f31a64a6e4f052/src/Native/Unix/System.Net.Security.Native/pal_gssapi.h#L47-L54

and

https://github.com/dotnet/corefx/blob/44e5bdafb8d989a220c9cf1b94f31a64a6e4f052/src/Native/Unix/System.Native/pal_networking.c#L844-L850

Reply:
> Currently there is only one place that defines these: this file. To be consistent with the current norm, should it be kept here?

I believe that the intention is that it will be deleted fairly soon, and if everything else is compiling without that flag then there won't be a second place.